core/src/debt.ts

```diff
@@ -0,0 +1,236 @@
+};
+
+export const restartDebtAuction = async function (
+    network: string,
+    auctionIndex: number,
+    notifier?: Notifier
+): Promise<void> {
+    await executeTransaction(network, 'MCD_FLOP', 'tick', [auctionIndex], { notifier });
+};
+
+export const bidToDebtAuction = async function (
+    network: string,
+    auctionIndex: number,
+    acceptableReceivedAmount: BigNumber,
+    notifier?: Notifier
+) {
+    const auction = (await getActiveDebtAuctionOrUndefined(network, auctionIndex)) as DebtAuctionActive | undefined;
+    if (!auction || !auction.receiveAmountMKR) {
+        throw new Error('Did not find the auction to bid on.');
+    }
+    const transactionParameters = [
+        auctionIndex,
+        acceptableReceivedAmount.shiftedBy(WAD_NUMBER_OF_DIGITS).toFixed(0),
+        auction.bidAmountDai.shiftedBy(RAD_NUMBER_OF_DIGITS).toFixed(0),
+    ];
+    await executeTransaction(network, CONTRACT, 'dent', transactionParameters, { notifier });
+};
+export const collectDebtAuction = async (network: string, auctionIndex: number, notifier?: Notifier) => {
+    const auction = await getActiveDebtAuctionOrUndefined(network, auctionIndex);
+    if (!auction || auction.state !== 'ready-for-collection') {
+        throw new Error('Did not find the auction to collect.');
+    }
+    await executeTransaction(network, 'MCD_FLOP', 'deal', [auctionIndex], { notifier });
+};
+export const enrichDebtAuction = async (
+    network: string,
+    auction: DebtAuctionActive
+): Promise<DebtAuctionTransaction> => {
+    const nextMaximumLotReceived = await getNextMaximumMkrReceived(network, auction);
+    const unitPrice = auction.bidAmountDai.div(auction.receiveAmountMKR);
+    const marketUnitPrice = await getMarketPriceDaiToMkr(network, auction.receiveAmountMKR);
+    const marketUnitPriceToUnitPriceRatio = unitPrice.minus(marketUnitPrice).dividedBy(marketUnitPrice);
+    const fees = await getDebtAuctionTransactionFees(network);
+    return {
+        ...auction,
+        ...fees,
+        nextMaximumLotReceived,
+        unitPrice,
+        marketUnitPrice,
+        marketUnitPriceToUnitPriceRatio,
+    };
+};
+
+export const enrichDebtAuctions = async (network: string, auctions: DebtAuction[]): Promise<DebtAuction[]> => {
+    const auctionsWithNextMinimumBidsPromises = auctions.map(auction => {
+        if (auction.state === 'collected') {
+            return auction;
+        }
+        return enrichDebtAuction(network, auction);
+    });
+    return await Promise.all(auctionsWithNextMinimumBidsPromises as Promise<DebtAuctionTransaction>[]);
+};
+
+export const getDebtAuctionLastIndex = async (contract: Contract): Promise<number> => {
+    const auctionsQuantityBinary = await contract.kicks();
+    return new BigNumber(auctionsQuantityBinary._hex).toNumber();
+};
+
+const _getDebtAuctionBidIncreaseCoefficient = async (network: string): Promise<BigNumber> => {
+    const contract = await getContract(network, 'MCD_FLOP');
```

Comment:
Do you want to replace all `MCD_FLAP` with the CONTRACT defined above for consistency? 💅